cmd/wego/app/add/cmd.go

```diff
@@ -82,27 +87,53 @@ func runCmd(cmd *cobra.Command, args []string) error {
 		params.Dir = path
 	}
 
+	var err error
 	osysClient := osys.New()
+	cliRunner := &runner.CLIRunner{}
+	fluxClient := flux.New(osysClient, cliRunner)
+	kubeClient := kube.New(cliRunner)
+	kube, rawClient, err := kube.NewKubeHTTPClient()
+	if err != nil {
+		return fmt.Errorf("error creating k8s http client: %w", err)
+	}
 
-	token, err := osysClient.GetGitProviderToken()
+	logger := logger.NewCLILogger(os.Stdout)
+	if err := app.IsClusterReady(logger, kube); err != nil {
+		return err
+	}
 
-	if err == osys.ErrNoGitProviderTokenSet {
-		// No provider token set, we need to do the auth flow.
-		url := params.Url
-		if url == "" {
-			// Find the url using an unauthenticated git client. We just need to read the URL.
-			// params.Dir must be defined here because we already checked for it above.
-			url, err = git.New(nil).GetRemoteUrl(params.Dir, "origin")
-			if err != nil {
-				return fmt.Errorf("could not get remote url for directory %s: %w", params.Dir, err)
-			}
+	isHelmChart := params.Chart != ""
```

Comment:
The isHelmChart could be a function on the Params struct itself.

Just thinking out loud here and this isn't really applicable to this PR.  However, it would be nice if `params` used embedded structs and we could return either one depending on whether we are dealing with `helm` or a `repo`.  This would simplify the code over the long-term.  Again, just a random thought outside of the scope of this PR.